December 2023 release notes (#1317)
semgrep/semgrep-docs: docs/release-notes/december-2023.md

hide_title: true
description: >-
    Release notes include the changes, fixes, and additions in specific versions of Semgrep.
toc_max_heading_level: 3
title: December 2023
tags:
  - Release notes
---

# December 2023 release notes

## 🔧 OSS Engine

* The following versions of the OSS Engine were released in December 2023:

  * [<i class="fas fa-external-link fa-xs"></i>
    1.52.0](https://github.com/returntocorp/semgrep/releases/tag/v1.52.0)
  * [<i class="fas fa-external-link fa-xs"></i>
    1.53.0](https://github.com/returntocorp/semgrep/releases/tag/v1.53.0)
  * [<i class="fas fa-external-link fa-xs"></i>
    1.54.0](https://github.com/returntocorp/semgrep/releases/tag/v1.54.0)
  * [<i class="fas fa-external-link fa-xs"></i>
    1.54.1](https://github.com/returntocorp/semgrep/releases/tag/v1.54.1)

## 🌐 Cloud Platform

### Added

* Semgrep IDE integrations now cache information about the repository so that it
  does not traverse the entire repository on every scan to determine if the
  files are valid targets for scanning; this improves scan times.
* Users can now ignore findings locally in Semgrep IDE Extensions. These changes
  persist between restarts.

> * Added authentication provider information for each user under **Settings** >

Reviewer: This was already there. We just moved the column (we only added it to the RBAC teams page which is still in private beta).

  **Access** > **Members** to help distinguish between users who log in with
  multiple methods.
* The metrics collected now include more granular information to help
differentiate scans using different engine capabilities, such as intraprocedural
scans without secrets validation versus intraprocedural scans *with* secrets
validation.
* **CLI tool**: Added new `semgrep test` subcommand, which is an alias for `semgrep scan
--test`. **Note**: When running Semgrep on a test directory, use `semgrep
scan test` to avoid confusion with the new `semgrep test` subcommand.

### Changed

> * Improved the speed with which Semgrep Cloud Platform's **Dashboard** and **Findings**

Reviewer: This is still being rolled out globally. I wonder if we should wait to put this into Jan's notes when everyone will have it?

Author: Will make a note to include in Jan

  pages load.
* Fixed issue where empty tables in `pyproject.toml` files would fail to parse.
* **OCaml**: Switched to a tree-sitter-based parser instead of the Menhir parser.
* **Rust**: Updated the parser used for Rust.

### Fixed

* Fixed issue where webhooks stopped working.
* Fixed issue so that clicking **Start Tour** now restarts the Getting Started
  tutorial.
* Fixed issue where the **Members** page doesn't display a user's new role until
  the page reloads. <!--OS-1233-->
* Fixed issue where users switching organizations would result in a 404.
  <!--OS-136-->
* Fixed the **Connect to** button under **Settings** > **Source Code Managers**
  so that it displays correctly based on whether the user can connect to a
  source code manager. <!-- https://github.com/semgrep/semgrep-app/pull/11812
  -->
* **CLI tool**: Updated CLI error message to clarify that users should log in before running
  either:
  * `semgrep ci`
  * `semgrep scan` then `semgrep scan --config`
  
## 💻 Code

### Fixed

* Fixed issue where Semgrep Code findings marked as **fixed** can be triaged via the rule

>   group. If the first finding in the list of findings is fixed, the rest

Reviewer: Not sure if this second sentence makes sense to me. I think just saying "if a rule is fixed, we can't change the triage status back to ignored" is enough?

  are also fixed; as such, the rule checkboxes are now disabled.
  <!--FIND-1453-->
* Fixed issue where the rule information card and the rule preview are missing
  for older findings; all findings now display this information.
  <!--FIND-1433-->
* Fixed issue where the finding's severity displayed doesn't match the rule's
  severity once the rule has been updated. <!--FIND-1397-->

> ## ⛓️ Supply Chain

Author: Need to remove this assuming I didn't miss anything

## 🤖 Assistant (beta)

### Added

* Added the **Analyze** button, which triggers all Assistant functions, including autofix,
autotriage, and component tagging, on selected findings. After Assistant performs these functions, users can see their results if they filter for findings based on **Recommendation** or by **Component**. Additionally, users that choose **No Grouping** instead of **Group by Rule** see false positive and true positive recommendations when viewing their finding details pages. 

## 🔐 Secrets (beta)

### Added

* Added support for custom validator rules.

### Changed

> * Disabled the **Add to Policy** button when writing custom validator rules for

Reviewer 2: This never worked, the UI is just kind of confusing right now so we disabled the button. I think this could be omitted

---
slug: december-2023
append_help_link: true
hide_title: true
description: >-
    Release notes include the changes, fixes, and additions in specific versions of Semgrep.
toc_max_heading_level: 3
title: December 2023
tags:
  - Release notes
---

> # December 2023 release notes

Reviewer 3: ```suggestion
# Semgrep release notes for December 2023
```

## 🔧 OSS Engine

* The following versions of the OSS Engine were released in December 2023:

  * [<i class="fas fa-external-link fa-xs"></i>
    1.52.0](https://github.com/returntocorp/semgrep/releases/tag/v1.52.0)
  * [<i class="fas fa-external-link fa-xs"></i>
    1.53.0](https://github.com/returntocorp/semgrep/releases/tag/v1.53.0)
  * [<i class="fas fa-external-link fa-xs"></i>
    1.54.0](https://github.com/returntocorp/semgrep/releases/tag/v1.54.0)
  * [<i class="fas fa-external-link fa-xs"></i>
    1.54.1](https://github.com/returntocorp/semgrep/releases/tag/v1.54.1)

## 🌐 Cloud Platform

### Added

> * Semgrep IDE integrations now cache information about the repository so that it

Reviewer 3: ```suggestion
* Semgrep IDE integrations now cache information about the current repository so that it
```

  does not traverse the entire repository on every scan to determine if the
  files are valid targets for scanning; this improves scan times.

> * Users can now ignore findings locally in Semgrep IDE Extensions. These changes

Reviewer 3: ```suggestion
* Users can now ignore findings locally in Semgrep IDE extensions. These changes
```

>   persist between restarts.

Reviewer 3: Hmm... "persist between restarts" but what happens when they commit + open the PR? Can you ask the engineer who worked on this what was the intent? I think it may be Austin.

Author: Per Austin:

> So when a user triages it in the ide, nothing is reported back to the app. This is just a local ignore that will persist between sessions. Nothing on the commit or PR

Reviewer 3: I'd suggest adding a note there stating this is a local ignore only. Something like "These changes persist between restarts, but are local ignores. Findings will still be detected when Semgrep scans your code, typically when opening a PR or MR.

(It's "typically" because a Semgrep scan, i.e. `semgrep scan` in the CLI will also generate findings.)

* The metrics collected now include more granular information to help
differentiate scans using different engine capabilities, such as intraprocedural
scans without secrets validation versus intraprocedural scans *with* secrets
validation.

> * **CLI tool**: Added new `semgrep test` subcommand, which is an alias for

Reviewer 3: Err, what's the difference between `semgrep test` and `semgrep scan test`?

Author: `semgrep scan test` indicates the repo name is `test` -- I can see how this is confusing. Would it be helpful to say something like, "Use `semgrep scan test` to explicitly name the test directory to avoid confusion...?"

Reviewer 3: Thanks, I have written a suggestion.

> `semgrep scan --test`. **Note**: When running Semgrep on a test directory, use

Reviewer 3: ```suggestion
`semgrep scan --test`. **Note**: If the **name** of the directory you are scanning is `test`, use `semgrep scan test` to avoid confusion with the new `semgrep test` subcommand.
```

`semgrep scan test` to avoid confusion with the new `semgrep test` subcommand.

### Changed

> * Fixed issue where empty tables in `pyproject.toml` files would fail to parse.

Reviewer 3: I suspect this is for Supply Chain, not SCP :)

* **OCaml**: Switched to a tree-sitter-based parser instead of the Menhir
  parser.

> * **Rust**: Updated the parser used for Rust.

Reviewer 3: Thank you for adding stuff like this.

### Fixed

> * Fixed issue where webhooks stopped working.

Reviewer 3: ```suggestion
* Fixed an issue where webhooks stopped working.
```
Note that we [prefer to include articles](https://developers.google.com/style/articles) for these types of sentences.

(Please apply to the rest of the release notes!)

* Fixed issue so that clicking **Start Tour** now restarts the Getting Started
  tutorial.
* Fixed issue where the **Members** page doesn't display a user's new role until
  the page reloads. <!--OS-1233-->
* Fixed issue where users switching organizations would result in a 404.
  <!--OS-136-->
* Fixed the **Connect to** button under **Settings** > **Source Code Managers**
  so that it displays correctly based on whether the user can connect to a
  source code manager. <!-- https://github.com/semgrep/semgrep-app/pull/11812
  -->
* **CLI tool**: Updated CLI error message to clarify that users should log in
  before running either:
  * `semgrep ci`

>   * `semgrep scan` then `semgrep scan --config`

Reviewer 3: ```suggestion
  * `semgrep scan` then `semgrep scan --config`
```
(Please hold - I'm testing this out for now)

  
## 💻 Code

### Fixed

> * Fixed issue where Semgrep Code findings marked as **fixed** can be triaged via

Reviewer 3: ```suggestion
* Fixed issue where Semgrep Code findings marked as **fixed** can be triaged through
```

>   the rule group. If the rule is fixed, the triage status cannot be changed back

Reviewer 3: ```suggestion
  the rule group. Once a finding is fixed, its triage status cannot be changed back
```

A **rule** isn't fixed, but a **finding** is.

  to **ignored**.
  <!--FIND-1453-->
* Fixed issue where the rule information card and the rule preview are missing
  for older findings; all findings now display this information.
  <!--FIND-1433-->
* Fixed issue where the finding's severity displayed doesn't match the rule's
  severity once the rule has been updated. <!--FIND-1397-->

## 🤖 Assistant (beta)

### Added

> * Added the **Analyze** button, which triggers all Assistant functions,

Reviewer 3: ```suggestion
* Added the **Analyze** button, which triggers all Assistant functions,
```
(Where is this button found? The settings page?)

including autofix, autotriage, and component tagging, on selected findings.
After Assistant performs these functions, users can see their results if they
filter for findings based on **Recommendation** or by **Component**.
Additionally, users that choose **No Grouping** instead of **Group by Rule** see
false positive and true positive recommendations when viewing their finding
details pages. 

## 🔐 Secrets (beta)

### Added

> * Added support for custom validator rules. Note that custom validator rules are

Reviewer 3: How can users make use of this feature, i.e. write their own validator rules? (I believe Sal is the engineer for this.)

Author: I think I added what you had in mind

append_help_link: true
hide_title: true
description: >-
    Release notes include the changes, fixes, and additions in specific versions of Semgrep.
toc_max_heading_level: 3
title: December 2023
tags:
  - Release notes
---

# Semgrep release notes for December 2023

## 🔧 OSS Engine

* The following versions of the OSS Engine were released in December 2023:

  * [<i class="fas fa-external-link fa-xs"></i>
    1.52.0](https://github.com/returntocorp/semgrep/releases/tag/v1.52.0)
  * [<i class="fas fa-external-link fa-xs"></i>
    1.53.0](https://github.com/returntocorp/semgrep/releases/tag/v1.53.0)
  * [<i class="fas fa-external-link fa-xs"></i>
    1.54.0](https://github.com/returntocorp/semgrep/releases/tag/v1.54.0)
  * [<i class="fas fa-external-link fa-xs"></i>
    1.54.1](https://github.com/returntocorp/semgrep/releases/tag/v1.54.1)

## 🌐 Cloud Platform

### Added

* Semgrep IDE integrations now cache information about the current repository so
  that it doesn't traverse the entire repository on every scan to determine if
  the files are valid targets for scanning; this improves scan times.
* Users can now ignore findings locally in Semgrep IDE Extensions. These changes

>   persist between restarts, though they're not reported back to Semgrep and do

Reviewer 3: ```suggestion
  persist between restarts, though they're not reported back to Semgrep Cloud Platform and do
```

Reviewer 3: I think this is one of those examples where the spelling out policy works, because technically the user HAS been using "Semgrep", but there's a difference between "local" Semgrep and "remote" Semgrep and we want to make that distinction clear.

>   not affect the remote repository or other users.

Reviewer 3: This is a minor nit, but I think it's worth it to spell out that btw, your ignored findings, if you don't change them, WILL appear again the next time a Semgrep scan is performed, because that's the end result to the user.

It's good we explain the behavior of the feature, but we also need to spell out HOW it affects the user.

---
slug: december-2023
append_help_link: true
hide_title: true
description: >-
    Release notes include the changes, fixes, and additions in specific versions of Semgrep.
toc_max_heading_level: 3
title: December 2023
tags:
  - Release notes
---

# Semgrep release notes for December 2023

## 🔧 OSS Engine

* The following versions of the OSS Engine were released in December 2023:

  * [<i class="fas fa-external-link fa-xs"></i>
    1.52.0](https://github.com/returntocorp/semgrep/releases/tag/v1.52.0)
  * [<i class="fas fa-external-link fa-xs"></i>
    1.53.0](https://github.com/returntocorp/semgrep/releases/tag/v1.53.0)
  * [<i class="fas fa-external-link fa-xs"></i>
    1.54.0](https://github.com/returntocorp/semgrep/releases/tag/v1.54.0)
  * [<i class="fas fa-external-link fa-xs"></i>
    1.54.1](https://github.com/returntocorp/semgrep/releases/tag/v1.54.1)

## 🌐 Cloud Platform

### Added

* Semgrep IDE integrations now cache information about the current repository so
  that it doesn't traverse the entire repository on every scan to determine if
  the files are valid targets for scanning; this improves scan times.

> * Users can now ignore findings locally in Semgrep IDE Extensions. The changes

Reviewer 3: ```suggestion
* Users can now ignore findings locally in Semgrep IDE extensions. The changes
```
;_; Does it keep eating my corrections?

Author: I think so; I double-checked this was there, too!
